macOS 12: TViewPubDataMembers uses deprecated std::iterator

```
In file included from src/core/meta/src/TClass.cxx:120:
src/core/meta/src/TViewPubDataMembers.h:97:45: warning: 'iterator<std::bidirectional_iterator_tag, TObject *, long, const TObject **, const TObject *&>' is deprecated [-Wdeprecated-declarations]
                                public std::iterator<std::bidirectional_iterator_tag, TObject *, std::ptrdiff_t,
                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__iterator/iterator.h:27:29: note: 'iterator<std::bidirectional_iterator_tag, TObject *, long, const TObject **, const TObject *&>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 iterator
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__config:1066:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#  define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__config:1043:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
                                               ^
```

--- a/core/meta/src/TViewPubDataMembers.h
+++ b/core/meta/src/TViewPubDataMembers.h
@@ -93,9 +93,14 @@ public:
 // Iterator of view of linked list.      `                              //
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
-class TViewPubDataMembersIter : public TIterator,
-                                public std::iterator<std::bidirectional_iterator_tag, TObject *, std::ptrdiff_t,
-                                                     const TObject **, const TObject *&> {
+class TViewPubDataMembersIter : public TIterator {
+public:
+   using iterator_category = std::bidirectional_iterator_tag;
+   using value_type = TObject *;
+   using difference_type = std::ptrdiff_t;
+   using pointer = const TObject **;
+   using reference =  const TObject *&;
+
 protected:
    const TList *fView;   //View we are iterating over.
    TIter        fClassIter;    //iterator over the classes

